### DISCUSSION TITLE
Unusual Checkbox Behaviour

### DISCUSSION THREAD

**Discussion by mdmcconnell** [pdfcpu/pdfcpu#993] [Q&A]
I think I have discovered another unsual use case of the PDF file format.  This is long set of related issues, and I think I have a (inelegant) workaround.  The attached file has many Yes/No checkboxes on page 5.  Though they appear as checkboxes when extracted to json, they behave like radio button sets.  In Acrobat, clicking a "Yes" will un-check its "No".  

I have been unable to set any "Yes" value with the pdfcpu api so started to investigate further.  I can check boxes using the CLI, and when I do that, they lose their "magic" nature (second file).  Also, only boxes with values appear in the form export of the filled pdf.  

I attempted to break the problem into smaller pieces using extract -m page, and another unusual change appeared: when I open a individual page of the form, now Acrobat gives me the message "Cannot Save Form Information - You cannot save a completed copy of this form on your computer".  (third file)

So the first question is "what is the nature of these Yes/No checkboxes that is so special?"
[f5471.pdf](https://github.com/user-attachments/files/17733179/f5471.pdf)
[f5471_all_checked.pdf](https://github.com/user-attachments/files/17733181/f5471_all_checked.pdf)
[f5471_checked_page_5.pdf](https://github.com/user-attachments/files/17733244/f5471_checked_page_5.pdf)

**Comment by mdmcconnell** [comment]
Ha! So, my original "problem" looks like an error in my program logic.  The curious (to me, at least) behaviour of these radio/checks remains.

**Comment by hhrutter** [comment]
Closed due to inactivity.
